app/views/spree/products/_cart_form.html.erb

```diff
@@ -85,6 +85,15 @@
 <% default_variant = default_variant(@variants, @product) %>
 <div class="product-side-details d-flex flex-column justify-content-between h-100"  >
   <div class="top-detail" >
+    <% s_hours=  @product.master.sale_prices.active.last.start_at.strftime("%H").to_i %>
+    <% e_hours= @product.master.sale_prices.active.last.end_at.strftime("%H").to_i %>
+    <% s_seconds=  @product.master.sale_prices.active.last.start_at.strftime("%S").to_i %>
+    <% e_seconds= @product.master.sale_prices.active.last.end_at.strftime("%S").to_i %>
+    <% s_date=  @product.master.sale_prices.active.last.start_at.strftime("%d").to_i %>
+    <% e_date=  @product.master.sale_prices.active.last.end_at.strftime("%d").to_i %>
+    <% if @product.on_sale? %>
+      <div class="deal-expiry-wrap"><strong>Deal Expiers in:</strong> <span><strong><%= e_date - s_date %></strong> day</span><span><strong><%= s_hours - e_hours %> </strong> hours</span><span><strong><%= s_seconds - e_seconds %></strong> seconds</span></div>
```

Comment:
this date is only shows in case of master variant if other variant is selected the what ?

Reply:
it is done sir